Assets/Scripts/Gameplay/Powerup/PowerupManager.cs

```diff
@@ -340,7 +376,7 @@ private void CheckForPowerupActivation()
 
         private bool CanActivatePowerUp()
         {
-            if (AnyPowerUp())
+            if (AnyPowerUp() || !_gameManager.GlobalGameReferences.LevelManager.HasReadWarning || _gameManager.GlobalGameReferences.IsFollowingSpline)

```

Comment:
Can this be split into bool variables to make it easier to read? Long if statement lines are not great to read.
`GlobalGameReferences globalGameRefs = _gameManager.GlobalGameReferences;`
`bool hasReadWarning = globalGameRefs.LevelManager.HasReadWarning`
`bool isFollowingSpline = globalGameRefs.IsFollowingSpline`

`bool powerupUseInvalid = !hasReadWarning || isFollowingSpline` (You can probably come up with a better name idea here 😂)

`if(AnyPowerup() || powerupUseInvalid) { return false; }`

Reply:
Also what is AnyPowerup() ? Just reading it without any context doesn't make sense to me, could this be renamed something a bit more readable and understandable?

Reply:
Declaring variables and storing them beforehand has the downside of using memory and clock cycles unnecessarily. A small amount but nonetheless inefficient. For the sake of readability, however, I've split the if statement onto multiple lines in the cs file.
I'll rename AnyPowerup to AnyPowerupActive though.